remove torch from package exporter

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72246
* __->__ #72242
* #72563
* #72237

Removes the dependency of torch on torch.package.package_exporter.

Differential Revision: [D33973987](https://our.internmc.facebook.com/intern/diff/D33973987)

PaliC committed:
remove torch from package exporter

[ghstack-poisoned]

PaliC committed:
Update on "remove torch from package exporter"




[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33973987).

suo commented on torch/package/package_exporter.py:
```diff
@@ -23,7 +23,7 @@
 )
 
 import torch
```
shouldn't we actually be removing torch lol

PaliC replied:
Indeed haha

suo commented on torch/package/package_exporter.py:
```diff
@@ -850,14 +850,6 @@ def deny(self, include: "GlobPattern", *, exclude: "GlobPattern" = ()):
 
     def _persistent_id(self, obj):
         if hasattr(obj, "__reduce_package__") and not inspect.isclass(obj):
-            if _gate_torchscript_serialization and isinstance(
```
if we are removing this check, shouldn't we remove the `_gate_torchscript_serialization` variable as well?

PaliC replied:
Yeah, I wanted to ask what was this for? Do we need to keep this in an internal version?

suo reviewed (changes requested):
some small cleanups but overall lgtm

PaliC committed:
Update on "remove torch from package exporter"


Differential Revision: [D33973987](https://our.internmc.facebook.com/intern/diff/D33973987)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33973987).

suo commented on torch/package/package_exporter.py:
```diff
@@ -690,7 +687,7 @@ def register_extern_hook(self, hook: ActionHook) -> RemovableHandle:
 
         Returns:
-            :class:`torch.utils.hooks.RemovableHandle`:
+            :class:`torch.package._hooks.RemovableHandle`:
```
This class should be a part of the public API, and thus should not have any underscores in its qualified name. The easiest way to do this is to re-export the name in `torch/pacakge/__init__.py` and reference it as `torch.package.RemovableHandle`.

suo reviewed (approved):
Small nit, otherwise lgtm

PaliC committed:
Update on "remove torch from package exporter"


Removes the dependency of torch on torch.package.package_exporter.

Differential Revision: [D33973987](https://our.internmc.facebook.com/intern/diff/D33973987)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33973987).
